File: c_solutions_1-10/Euler_10.c
#include "euler_util.h"


int main(int argc, char *argv[])
{
    float start = timeit();

    unsigned long long total;
    unsigned long i;

    for (i=0, total=0; i < 2000000; i++)
        total += is_prime(i) ? i : 0;

    float stop = timeit();

    printf("Answer: %lld\n", total);
    printf("Time: %.8f\n", stop - start);

    return 0;
}

File: perl_solutions_1-10/Euler_10.pl
use strict;
use warnings;
use Math::Complex;
use Time::HiRes qw( clock );


sub is_prime
{
    my ($n) = @_;
    if ($n == 2) {
        return 1;
    }
    if ($n % 2 == 0 || $n == 1) {
        return 0;
    }
    for (my $i=3; $i < int(sqrt($n)) + 1; $i++) {
        if ($n % $i == 0) {
            return 0;
        }
    }
    return 1;
}

sub all_primes
{
    my ($limit) = @_;
    my $total = 0;
    for (my $i=2; $i < $limit; $i++) {
        if ((is_prime($i))) {
            $total += $i
        }
    }
    return $total;
}

my $start = clock();
my $answer = all_primes(2000000);
my $stop = clock();
print "Answer: ", $answer, "\n";
printf "Time: %.5f\n", $stop - $start;
